### ISSUE TITLE
Accessibility: Add label for screen readers to close button

### ISSUE THREAD

**Issue by laurelfulford** [Automattic/republication-tracker-tool#134]
This issue came up in a accessibility review of one of Newspack's publishers' sites:

The close button in the modal dialogue should have a label of some kind that can be used by screen readers (eg. text wrapped with the `screen-reader-text` class, and the appropriate styles added to the plugin in cases where a site isn't using a theme that already has them. 

Related resource: https://www.w3.org/TR/UNDERSTANDING-WCAG20/navigation-mechanisms-descriptive.html
